### DISCUSSION TITLE
How can vulnerability sources be updated within docker-compose.yml ?

### DISCUSSION THREAD

**Discussion by mciccarone** [DependencyTrack/dependency-track#2375] [Q&A] [answered]
I am trying to figure out how to configure the DependencyTrack apiserver to update it's default NVD vulnerability feed to reference another docker-container that is hosting an offline clone of the NIST vulnerabilities.

`docker-compose.yml contains:   [file-server]    [dtrack-apiserver]    [dtrack-frontend]`

My configuration has the dependency-track apiserver and the file-server in the same docker-compose.yml file so that they can communicate with one another without getting into advanced network configuration of multiple docker compositions.  My offline file server for NIST data is simply named "file-server" hosting a file server over port 8000. When I run DependencyTrack frontend I can manually update the NIST feeds to file-server and upon restarting the apiserver it will start grabbing the files from my offline file-server.

I think the problem is that the configuration for source vulnerabilities is contained in a java enum containing multiple values `[string:group] [string:url] [string:bool] [PropertyTpe] [string:description]` objects. I have no idea how to configure these enumation sets in the compose file.

I have tried passing in all of these variables into the docker-compose apiserver environment, but nothing seems to have any effect:

environment:
    - VULNERABILITY_SOURCE_NVD_FEEDS_URL=http://file-server:8000/nist/feeds
    - nvd.feeds.url='http://file-server:8000/nist/feeds'
    - NVD_FEEDS_URL=http://file-server:8000/nist/feeds
    - "JAVA_OPTS=-XX:NativeMemoryTracking=summary -Dnvd.feeds.url=http://file-server:8000/nist/feeds/"

**Does anyone know how to set these configuration items during the docker composition phase without having to rely on the frontend to modify the URLs ?**

**Comment by syalioune** [comment, marked as answer]
Hello @mciccarone 

You can only set those values using the UI or the REST API. For the REST API :

- Change the default admin password, login and get token following https://github.com/DependencyTrack/dependency-track/issues/831#issue-741441433
- Use the `POST /v1/configProperty {"groupName":"..","propertyName":"...","propertyValue":"..."}` endpoint

I'm not aware of any way to execute a command after container is ready using docker compose. If I were in your shoes, I'll setup another container with a `depends_on` relationship with `dtrack-apiserver` and have it run my init script.
